ui/artalk/src/plugins/page-vote.ts

```diff
@@ -0,0 +1,75 @@
+import ActionBtn from '../components/action-btn'
+import type { ArtalkPlugin, ContextApi } from '@/types'
+import $t from '@/i18n'
+
+interface PageVoteOptions {
+  pageKey: string
+  vote: boolean
+  voteDown: boolean
+  btnEl: string
+}
+
+export const PageVoteWidget: ArtalkPlugin = (ctx) => {
+  ctx.watchConf(['pageKey', 'pageVote', 'darkMode'], (conf) => {
+    if (!conf.pageVote) return
+
+    const defaultOptions = {
+      pageKey: conf.pageKey,
+      vote: true,
+      voteDown: false,
+      btnEl: '.artalk-page-vote',
+    }
+
+    const pageVoteConfig = typeof conf.pageVote === 'object' ? conf.pageVote : {}
+    initPageVoteWidget(ctx, { ...defaultOptions, ...pageVoteConfig })
+  })
+}
+
+function initPageVoteWidget(ctx: ContextApi, options: PageVoteOptions) {
+  const btnContainer = document.querySelector<HTMLElement>(options.btnEl)
+  if (!btnContainer) return // throw Error(`page vote's config \`btnEl\` selector ${options.btnEl} not found`)
```

Comment:
to avoid [test failed](https://github.com/ArtalkJS/Artalk/actions/runs/10971071431/job/30467155651?pr=983#step:8:80), it skips when enable page vote but there is no element to render, is this ok?

Reply:
Yes, if the relevant elements are not found, just disable this feature directly. This is the same logic as the configuration of `pvEl` and `countEl`.